components/TextField/TextField.tsx

```diff
@@ -118,6 +141,11 @@ export const TextField = React.forwardRef<React.ElementRef<typeof Input>, TextFi
       [hasAdornmentGroup]
     );
 
+    const PasswordVisibilityToggleIcon = React.useMemo(
+      () => isPasswordVisible ? StyledEyeClosedIcon : StyledEyeOpenIcon,
```

Comment:
I feel we should display a closed eye when the password is not visible, and an open eye when the password is visible.

```suggestion
      () => isPasswordVisible ? StyledEyeOpenIcon : StyledEyeClosedIcon,
```

Reply:
@paulocfjunior I expect both cases to be fine.

The matter has never really been solved since timestamps exist : https://ux.stackexchange.com/questions/54162/icon-for-unmasking-passwords-open-or-closed-to-begin-with/54168

I expect the best way would be to add tooltips to make sure the action is understandable.

I don't have a strong opinion on which icon should match which state, though, so I'll apply your recommendation.

Reply:
I agree, didn't know it was such an old discussion. 🤯

Reply:
I quite agree with Paulo it looks weird to me to have closed eyes when the password is hidden. On the StackExchange link, you are shared I would go on what is said in the message having the most up votes. IMO a button should display a label or icon related to the action it will do.
UX is about user habits so checking what is the common usage is a good way to see what users would expect most. https://dribbble.com/search/password%20input we can see that most designers are putting opened eyes as default.

Reply:
For the record, @matthieuh is asking the opposite of @paulocfjunior.
To me, the last argument makes sense - follow user habits - plus I'm also used to seeing icon related to the consequence of action rather than current state, when talking about password inputs.
I think the tooltip helps solving the confusion overall. We're used to having the ability to show/hide password, hence the icon won't have a huge impact on the interaction for users.